From fb4e6985e142da56bad143d70600cd3695c91757 Mon Sep 17 00:00:00 2001
From: Brian Cain <bcain@hashicorp.com>
Date: Wed, 11 Dec 2019 13:25:11 -0800
Subject: [PATCH] Fixes #11249: Add VirtualBox provider support for version
 6.1.x

This commit adds support for VirtualBox version 6.1.x. It simply
inherits from the base 6.0.x provider class.
---
 plugins/providers/virtualbox/driver/meta.rb      |  1 +
 .../providers/virtualbox/driver/version_6_1.rb   | 16 ++++++++++++++++
 plugins/providers/virtualbox/plugin.rb           |  1 +
 website/source/docs/virtualbox/index.html.md     |  2 +-
 4 files changed, 19 insertions(+), 1 deletion(-)
 create mode 100644 plugins/providers/virtualbox/driver/version_6_1.rb

diff --git a/plugins/providers/virtualbox/driver/meta.rb b/plugins/providers/virtualbox/driver/meta.rb
index 24547b4044..ed013737b0 100644
--- a/plugins/providers/virtualbox/driver/meta.rb
+++ b/plugins/providers/virtualbox/driver/meta.rb
@@ -64,6 +64,7 @@ def initialize(uuid=nil)
             "5.1" => Version_5_1,
             "5.2" => Version_5_2,
             "6.0" => Version_6_0,
+            "6.1" => Version_6_1,
           }
 
           if @@version.start_with?("4.2.14")
diff --git a/plugins/providers/virtualbox/driver/version_6_1.rb b/plugins/providers/virtualbox/driver/version_6_1.rb
new file mode 100644
index 0000000000..ebe417eb91
--- /dev/null
+++ b/plugins/providers/virtualbox/driver/version_6_1.rb
@@ -0,0 +1,16 @@
+require File.expand_path("../version_6_0", __FILE__)
+
+module VagrantPlugins
+  module ProviderVirtualBox
+    module Driver
+      # Driver for VirtualBox 6.1.x
+      class Version_6_1 < Version_6_0
+        def initialize(uuid)
+          super
+
+          @logger = Log4r::Logger.new("vagrant::provider::virtualbox_6_1")
+        end
+      end
+    end
+  end
+end
diff --git a/plugins/providers/virtualbox/plugin.rb b/plugins/providers/virtualbox/plugin.rb
index e058ec65ad..f2fbf47e55 100644
--- a/plugins/providers/virtualbox/plugin.rb
+++ b/plugins/providers/virtualbox/plugin.rb
@@ -59,6 +59,7 @@ module Driver
       autoload :Version_5_1, File.expand_path("../driver/version_5_1", __FILE__)
       autoload :Version_5_2, File.expand_path("../driver/version_5_2", __FILE__)
       autoload :Version_6_0, File.expand_path("../driver/version_6_0", __FILE__)
+      autoload :Version_6_1, File.expand_path("../driver/version_6_1", __FILE__)
     end
 
     module Model
diff --git a/website/source/docs/virtualbox/index.html.md b/website/source/docs/virtualbox/index.html.md
index 558ef5f5d8..e9c1e79aad 100644
--- a/website/source/docs/virtualbox/index.html.md
+++ b/website/source/docs/virtualbox/index.html.md
@@ -13,7 +13,7 @@ Vagrant comes with support out of the box for [VirtualBox](https://www.virtualbo
 a free, cross-platform consumer virtualization product.
 
 The VirtualBox provider is compatible with VirtualBox versions 4.0.x, 4.1.x,
-4.2.x, 4.3.x, 5.0.x, 5.1.x, 5.2.x, and 6.0.x. Other versions are unsupported and the provider
+4.2.x, 4.3.x, 5.0.x, 5.1.x, 5.2.x, 6.0.x, and 6.1.x. Other versions are unsupported and the provider
 will display an error message. Please note that beta and pre-release versions
 of VirtualBox are not supported and may not be well-behaved.
 
